main.tf

```diff
@@ -20,7 +16,8 @@ resource "azurerm_management_group_policy_assignment" "this" {
 
   management_group_id  = local.management_group_id
   policy_definition_id = each.value.policy_definition_id
-  parameters           = jsonencode(each.value.parameters)
+  #If parameters is an empty map, pass null
+  parameters = length(each.value.parameters) == 0 ? null : jsonencode(each.value.parameters)
```

Comment:
minor: align line 20 with those above:
```
  management_group_id  = local.management_group_id
  policy_definition_id = each.value.policy_definition_id
  parameters           = length(each.value.parameters) == 0 ? null : jsonencode(each.value.parameters)

Reply:
if there's a comment above `terraform fmt` won't mark is a formatting error, since it's not a single block anymore. If I do manual alignment, then the hook will throw an error

Reply:
right, then we can ignore it